lib/tracker/request-tracker.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+/**
+ * Tracks request execution for a {@link Client}.
+ * <p>
+ *   A {@link RequestTracker} can be configured in the query options. The <code>Client</code> will execute
+ *   {@link Client#onSuccess} or {@link Client#onError} for every query or batch executed (QUERY, EXECUTE and BATCH
+ *   requests).
+ * </p>
+ * @interface
+ * @alias module:tracker~RequestTracker
+ */
+class RequestTracker {
+
+  /**
+   * Invoked each time a query or batch request succeeds.
+   * @param {Host} host The node that acted as coordinator of the request.
+   * @param {String|Array} query In the case of prepared or unprepared query executions, the provided
+   * query string. For batch requests, an Array containing the queries and parameters provided.
+   * @param {Array|Object|null} parameters In the case of prepared or unprepared query executions, the provided
+   * parameters.
+   * @param {QueryOptions} queryOptions The query options used to execute the request.
+   * @param {Number} requestLength Length of the body of the request.
+   * @param {Number} responseLength Length of the body of the response.
+   * @param {Array<Number>} latency An array containing [seconds, nanoseconds] tuple, where nanoseconds is the
+   * remaining part of the real time that can't be represented in second precision (see <code>process.hrtime()</code>).
+   */
+  onSuccess(host, query, parameters, queryOptions, requestLength, responseLength, latency) {
+    throw new Error('Not implemented');
```

Comment:
This is a small thing, but what do you think about having these do nothing instead of throwing errors?  That way implementations only need to  implement one of `onSuccess` / `onError` .

Reply:
I think it makes more sense to avoid throwing an error as its likely that a user would like to implement just 1 of them ✔️